habitat_baselines/evaluate_ppo.py

```diff
@@ -149,13 +195,151 @@ def main():
         episode_counts += 1 - not_done_masks
         current_episode_reward *= not_done_masks
 
+        next_episodes = envs.current_episodes()
+        envs_to_pause = []
+        n_envs = envs.num_envs
+        for i in range(n_envs):
+            if next_episodes[i].episode_id in stats_episodes:
+                envs_to_pause.append(i)
+
+            # episode ended
+            if not_done_masks[i].item() == 0:
+                stats_episodes.add(current_episodes[i].episode_id)
+                if args.video_option:
+                    generate_video(
+                        args,
+                        rgb_frames[i],
+                        current_episodes[i].episode_id,
+                        cur_ckpt_idx,
+                        infos[i]["spl"],
+                        writer,
+                    )
+                    rgb_frames[i] = []
+
+            # episode continues
+            elif args.video_option:
+                frame = observations_to_image(observations[i], infos[i])
+                rgb_frames[i].append(frame)
+
+        # stop tracking ended episodes if they exist
+        if len(envs_to_pause) > 0:
+            state_index = list(range(envs.num_envs))
+            for idx in reversed(envs_to_pause):
+                state_index.pop(idx)
+                envs.pause_at(idx)
+
+            # indexing along the batch dimensions
+            test_recurrent_hidden_states = test_recurrent_hidden_states[
+                :, state_index
+            ]
+            not_done_masks = not_done_masks[state_index]
+            current_episode_reward = current_episode_reward[state_index]
+
+            for k, v in batch.items():
+                batch[k] = v[state_index]
+
+            if args.video_option:
+                rgb_frames = [rgb_frames[i] for i in state_index]
+
     episode_reward_mean = (episode_rewards / episode_counts).mean().item()
     episode_spl_mean = (episode_spls / episode_counts).mean().item()
     episode_success_mean = (episode_success / episode_counts).mean().item()
 
-    print("Average episode reward: {:.6f}".format(episode_reward_mean))
-    print("Average episode success: {:.6f}".format(episode_success_mean))
-    print("Average episode spl: {:.6f}".format(episode_spl_mean))
+    logger.info("Average episode reward: {:.6f}".format(episode_reward_mean))
```

Comment:
To deduplicate logging we even can add option`log=True` to `writer.add_scalars` method and log scalars from TensorBoard Wrapper.